### DISCUSSION TITLE
Notes and music desynchronized after playing for a while

### DISCUSSION THREAD

**Discussion by yoyyoy** [ppy/osu#30552] [Q&A]
I left a comment on https://github.com/ppy/osu/issues/23992 a couple weeks ago but didn't get any response. don't wanna make duplicate issue so decided to go here.

I've had this issue happen to me and https://github.com/ppy/osu/pull/24449 initially fixed it for me, but this has recently resurfaced. I switched from plugging my headphones from into my monitor to into my motherboard. I did this because i recently heard about REAL.exe and wanted to take advantage of it, but the nvidia audio drivers don't support smaller buffer sizes than the default 10ms so REAL.exe does nothing when i connect my headphones to my monitor. as soon as i switched i noticed the desync and was reminded of the issues i faced before that clock fix.

when the game is out of sync, setting the audio output device from default, to my headphones fixes it temporarily. despite my headphones already being the default. setting the output to my monitor I was unable to get it to happen

as for fixes I tried, I've tested with and without using REAL.exe and it happens in both cases. i've tried updating my audio driver but it was already up to date. I tried what asus provides for my motherboard, some realtek driver, but that caused really high audio latency so i immediately went back to windows default driver. I noticed after i uninstalled the realtek driver, which required a restart, i had no issues for the rest of the day. but today it happened again. maybe putting my computer to sleep caused issues.

it may very well be a driver issue, but considering most people are probably using windows default driver I feel it's worthwhile to try and at least mitigate it. I've seen other users with similar behavior, like https://github.com/ppy/osu/issues/26237 or https://github.com/ppy/osu/issues/28595, plus, i've never seen this behavior in stable so it must be something unique to lazer. I really don't want to go back to stable so i just haven't been playing in the meantime

**Comment by peppy** [comment]
If you're still encountering this, please read https://github.com/ppy/osu/issues/31843 and test the provided fix if you're able!
